internal/provider/alb/pool_resource.go

```diff
@@ -0,0 +1,468 @@
+		GracefulTimeoutPeriod:    utils.TakeIntPointer(int(d.GracefulTimeoutPeriod.ValueInt64())),
+		PassiveMonitoringEnabled: d.PassiveMonitoringEnabled.ValueBoolPointer(),
+	}
+
+	poolMembers, err := r.getAlbPoolMembersType(d)
+	if err != nil {
+		return nil, fmt.Errorf("error defining pool members: %w", err)
+	}
+	albPoolConfig.Members = poolMembers
+
+	persistenceProfile, err := r.getAlbPoolPersistenceProfileType(d)
+	if err != nil && !errors.Is(err, ErrPersistenceProfileIsEmpty) {
+		return nil, fmt.Errorf("error defining persistence profile: %w", err)
+	}
+	albPoolConfig.PersistenceProfile = persistenceProfile
+
+	healthMonitors, err := r.getAlbPoolHealthMonitorType(d)
+	if err != nil {
+		return nil, fmt.Errorf("error defining health monitors: %w", err)
+	}
+	albPoolConfig.HealthMonitors = healthMonitors
+
+	return albPoolConfig, nil
+}
+
+// getAlbPoolMembersType.
+func (r *albPoolResource) getAlbPoolMembersType(d *albPoolModel) ([]govcdtypes.NsxtAlbPoolMember, error) {
+	var members []member
+	diags := d.Members.ElementsAs(context.Background(), &members, true)
+	if diags.HasError() {
+		return nil, errors.New(diags[0].Detail())
+	}
+	memberSlice := make([]govcdtypes.NsxtAlbPoolMember, 0)
+	for _, memberDefinition := range members {
+		memberSlice = append(memberSlice, govcdtypes.NsxtAlbPoolMember{
+			Enabled:   memberDefinition.Enabled.ValueBool(),
+			IpAddress: memberDefinition.IPAddress.ValueString(),
+			Ratio:     utils.TakeIntPointer(int(memberDefinition.Ratio.ValueInt64())),
+			Port:      int(memberDefinition.Port.ValueInt64()),
+		})
+	}
+	return memberSlice, nil
+}
+
+// getAlbPoolPersistenceProfileType.
+func (r *albPoolResource) getAlbPoolPersistenceProfileType(d *albPoolModel) (*govcdtypes.NsxtAlbPoolPersistenceProfile, error) {
+	if d.PersistenceProfile.IsNull() {
+		return nil, ErrPersistenceProfileIsEmpty
+	}
+
+	p := &persistenceProfile{}
+	diags := d.PersistenceProfile.As(context.Background(), p, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    false,
+		UnhandledUnknownAsEmpty: false,
+	})
+	if diags.HasError() {
+		return nil, errors.New(diags[0].Detail())
+	}
+
+	persistenceProfile := &govcdtypes.NsxtAlbPoolPersistenceProfile{}
+	persistenceProfile.Type = p.Type.ValueString()
+	persistenceProfile.Value = p.Value.ValueString()
+
+	return persistenceProfile, nil
+}
+
+// getAlbPoolHealthMonitorType.
+func (r *albPoolResource) getAlbPoolHealthMonitorType(d *albPoolModel) ([]govcdtypes.NsxtAlbPoolHealthMonitor, error) {
+	var healthMonitors []string
+	diags := d.HealthMonitors.ElementsAs(context.Background(), &healthMonitors, true)
+	if diags.HasError() {
+		return nil, errors.New(diags[0].Detail())
+	}
+
+	healthMonitorSlice := make([]govcdtypes.NsxtAlbPoolHealthMonitor, 0)
+
+	for _, healthMonitor := range healthMonitors {
+		healthMonitorSlice = append(healthMonitorSlice, govcdtypes.NsxtAlbPoolHealthMonitor{
+			Type: healthMonitor,
+		})
+	}
+
+	return healthMonitorSlice, nil
+}
```

Comment:
Passing `context.Background()` in the functions `As` and `ElementsAs` is working correctly ?